src/dashboard/dashboard.js

```diff
@@ -1,10 +1,59 @@
 import React from "react";
+import ChatListComponent from "../chatlist/chatList";
+import firebase from "../firebase";
 
 class DashboardComponent extends React.Component {
-    
-    render() {
-        return(<div>Hello World from Dashboard</div>)
-    }
+  constructor() {
+    super();
+    this.state = {
+      selectedChat: null,
+      newChatFormVisible: false,
+      email: null,
+      chats: [],
+    };
+  }
+
+  selectChat = (chatIndex) => {
+    console.log("Selected a chat!", chatIndex);
+  };
+
+  newChatBtnClicked = () => {
+    this.setState({ newChatFormVisible: true, selectedChat: null });
+  };
+
+  componentDidMount = () => {
+    firebase.auth().onAuthStateChanged(async (_usr) => {
+      if (!_usr) {
+        this.props.history.push("/login");
+      } else {
+        await firebase
+          .firestore()
+          .collection("chats")
+          .where("users", "array-contains", _usr.email)
+          .onSnapshot(async (res) => {
+            const chats = res.docs.map((_doc) => _doc.data());
```

Comment:
This is where I was getting stuck! (which I mentioned in my email)... `chats` was always an empty array for me 😔
Hopefully we'll be able to resolve it so I can see what all this wonderful code looks like in the browser!

Reply:
I sent you the details for my own database. I think it will not be a problem anymore:)